### PR TITLE
Mean power detection

### PR DESCRIPTION
**Pull request by MichaelCotton**
I cleaned some stuff up and added a `Detection` object to support lab y factor cals. @djanderson gave it a first review. @linhvu1 and @bradeales : Pls give it a look.  I hope this gets @linhvu1 one step closer to using scos-sensor.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,28 +198,51 @@ Per SigMF, the annotations value is an array of annotation segment objects that
 #### 4.3.1 Measurement Types
 The following annotation objects are used within the `scos` SigMF name space associated with `measurement_type`. 
 
-##### SingleFrequencyFFTDetection Object
-Single-frequency FFT detection is a standard software-defined radio measurement. The `SingleFrequencyFFTDetection` object contains the following name/value pairs:  
+##### Detection Object
```
Comment (bradeales) [inline README.md]: What is the purpose of this object? How does it differ from all the other detection objects? Do you mean it is a generic or basic detection algorithm? Maybe call it "BasicDetection Object" or something like that? As "Detection Object", it makes it seem it's a parent object to all the others.

File: README.md [hunk 2]
```diff
@@ -184,7 +185,7 @@ Per SigMF, the annotations value is an array of annotation segment objects that
 |----|--------------|-------|-------|-----------|
 |`altitude`|false|float|meters|The height of the antenna above mean sea level.|
 |`environment`|false|string|N/A|A description of the environment where antenna is mounted. E.g. `"indoor"` or `"outdoor"`.|
-|`measurement_type`|true|object|N/A|The type of measurement acquired: [SingleFrequencyFFTDetection](#singlefrequencyfftdetection-object), [SteppedFrequencyFFTDetection](#steppedfrequencyfftdetection-object), [SweptTunedDetection](#swepttuneddetection-object) or [YFactorCalibration](#yfactorcalibration-object).|
+|`measurement_type`|true|object|N/A|The type of measurement acquired, e.g., [SteppedFrequencyDetection](#steppedfrequencydetection-object), [SweptTunedDetection](#swepttuneddetection-object) or [YFactorCalibration](#yfactorcalibration-object).|
```
Comment (bradeales) [inline README.md]: Do you want to list all the new detection types here as well?

File: README.md [hunk 3]
```diff
@@ -104,8 +105,8 @@ The `Antenna` object contains the following name/value pairs:
 |`low_frequency`|false|float|Hz|Low frequency of operational range.|
 |`high_frequency`|false|float|Hz|High frequency of operational range.|
 |`gain`|false|float|dBi|Antenna gain in direction of maximum radiation or reception.|
-|`horizontal_gain_pattern`|false|array|dBi|Antenna gain pattern in horizontal plane.|
-|`vertical_gain_pattern`|false|array|dBi|Antenna gain pattern in vertical plane.|
+|`horizontal_gain_pattern`|false|array of floats|dBi|Antenna gain pattern in horizontal plane from 0 to 359 degrees in 1 degree steps.|
```
Comment (bradeales) [inline README.md:89]: `float` is not needed for 1 degree steps as per description. "array of integers" perhaps?

Comment (bradeales) [reply README.md:89]: Also, why do we need an array, is it not just a list?

File: README.md [hunk 4]
```diff
@@ -104,8 +105,8 @@ The `Antenna` object contains the following name/value pairs:
...
+|`vertical_gain_pattern`|false|array of floats|dBi|Antenna gain pattern in vertical plane from -90 to +90 degrees in 1 degree steps.|
```
Comment (bradeales) [inline README.md:90]: Do we need an array if we specify in the description the start, end, and increment? List?

File: README.md (hunk 3 above)
Comment (MichaelCotton) [reply README.md:89]: Floats are needed for antenna gain numbers. The 0 to 359 count describe the azimuth angles to which the gains correspond.

**Review by bradeales — CHANGES_REQUESTED** [review]
Biggest question is about new "Detection Object".

File: README.md (hunk 1 above)
Comment (MichaelCotton) [reply README.md]: I have been struggling with this. The purpose is to provided metrics on the time series. Detection is a stretch, but I don't know what else to call it. Perhaps TimeSeriesCalculations?

Comment (MichaelCotton) [reply README.md]: Instead of `Detection` v `FFTDetection`, `TimeDomainDetection` v `FrequencyDomainDetection`?

File: README.md (hunk 2 above)
Comment (MichaelCotton) [reply README.md]: The new detection types are a subsets of `SteppedFrequencyDetection`. Basically, if you specify one frequency then it reduces to new detection types.

**Review by bradeales — APPROVED** [review]

**Comment by bradeales** [comment]
I like it :+1:. Every question I could think of seems to be covered!

**Review by linhvu1 — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -146,20 +128,22 @@ Each `RFPath` object contains the following name/value pairs:
...
+|`altitude`|false|float|meters|Altitude above mean sea level.|
+
+### 3.3 ScheduleEntry Object
+The `ScheduleEntry` object is associated with the SCOS control plane, which is implemented through the use of a RESTful API residing on the sensor, see [SCOS Sensor](https://github.com/NTIA/scos-sensor) and [SCOS Control Plane API Reference](https://ntia.github.io/scos-sensor/). A sensor advertises its **capabilities**, among which are **actions** that users can schedule the sensor to do. Sensor actions are scheduled by posting a **schedule entry** to the sensor's **schedule**. The scheduler periodically reads the schedule and populates a task queue in priority order. A **task** represents an action at a _specific_ time. Therefore, a schedule entry represents a range of tasks. The scheduler continues populating its task queue until the schedule is exhausted. When executing the task queue, the scheduler makes a best effort to run each task at its designated time, but the scheduler SHOULD NOT in most cases cancel a running task to start another task, even of higher priority. **priority** is used to disambiguate two or more tasks that are schedule to start at the same time.
```
Comment (djanderson) [inline README.md]: "are schedule to" -> "are scheduled to" in the last sentence.

File: README.md [hunk 6]
```diff
@@ -174,54 +158,116 @@ The `ScheduleEntry` object contains the following name/value pairs:
...
+  "frequencies": [100000000, 200000000, 300000000, 400000000, 500000000]
+  "algorithm"
+  {
+    "detector": "mean_power",
+    "number_of_samples": 100000,
+    "units": "dBm",
+    "reference": "output of isotropic antenna"
+  }
+}
+```
```
Comment (djanderson) [inline README.md:268]: 1) This isn't valid JSON (missing comma after frequencies array, missing colon after algorithm). I recommend dropping any JSON examples into an online JSON validator/formatter just to make sure I didn't miss anything as well. 

2) I think this is trying to be just a bit too concise to the point of making it difficult to interpret. Remember that when you use a `TimeDomainDetection` object or a `FrequencyDomainDetection` object, they're not actually labeled as such. So I'm looking at this and thinking "okay so to figure out if this is a time domain or frequency domain stepped measurement, I'm going to need to look at the different fields in both objects, pick a field that's required in one but doesn't exist in the other and look for that"... etc.

Quick off-the-cuff suggestion: `detection_domain: "time"|"frequency"`?

**Review by djanderson — CHANGES_REQUESTED** [review]

**Review by djanderson — APPROVED** [review]

**Pull request merged**